app/validators/date_format_validator.rb

```diff
@@ -1,12 +1,26 @@
 class DateFormatValidator < ActiveModel::Validator
   def validate(record)
     options[:fields].each do |field|
-      date = record.send(field)
-      next if date.blank?
+      validate_date_fields(field, record)
+    end
+  end
 
-      match = date.strftime('%Y-%m-%d').match(/^(?<y>\d*)\-/)
+  private
 
-      record.errors.add(field, I18n.t('errors.messages.year_invalid')) if match[:y].length > 4
+  def validate_date_fields(field, record)
+    day = record.send(field.to_s + '_dd')
+    month = record.send(field.to_s + '_mm')
+    year = record.send(field.to_s + '_yyyy')
+    return if day.blank? || month.blank? || year.blank?
+    begin
+      Date.parse("#{day}-#{month}-#{year}")
+    rescue
+      record.errors.add(field, invalid_field_error(field))
     end
+    record.errors.add(field, I18n.t('errors.messages.year_invalid')) if year.length > 4
```

Comment:
I'm wondering if we should do more to validate the year? I.e. 2050 is a valid year but realistically should we be allowing that?

Reply:
I wasn't trying to keep the current behaviour since this ticket is about the content update, but if you think its worth to review this, I can get some input for the team and do it in a separate ticket :)

Reply:
Might just be worth chucking into an icebox :)

Reply:
https://dfedigital.atlassian.net/secure/RapidBoard.jspa?rapidView=21&view=planning&selectedIssue=TEVA-277